kuma/wiki/forms.py

```diff
@@ -327,63 +356,57 @@ def clean_current_rev(self):
             # If there's no document yet, just bail.
             return current_rev
 
-    def save_section(self, creator, document, **kwargs):
-        """Save a section edit."""
-        # This is separate because the logic is slightly different and
-        # may need to evolve over time; a section edit doesn't submit
-        # all the fields, and we need to account for that when we
-        # construct the new Revision.
-
-        old_rev = Document.objects.get(pk=self.instance.document.id).current_revision
-        new_rev = super(RevisionForm, self).save(commit=False, **kwargs)
-        new_rev.document = document
-        new_rev.creator = creator
-        new_rev.toc_depth = old_rev.toc_depth
-        new_rev.save()
-        new_rev.review_tags.set(*list(old_rev.review_tags
-                                             .values_list('name', flat=True)))
-        return new_rev
-
-    def save(self, creator, document, **kwargs):
-        """Persist me, and return the saved Revision.
-
-        Take several other necessary pieces of data that aren't from the
-        form.
-
+    def save(self, request, document, **kwargs):
         """
-        if (self.section_id and self.instance and
-                self.instance.document):
-            return self.save_section(creator, document, **kwargs)
-        # Throws a TypeError if somebody passes in a commit kwarg:
-        new_rev = super(RevisionForm, self).save(commit=False, **kwargs)
-
-        new_rev.document = document
-        new_rev.creator = creator
-        new_rev.toc_depth = self.cleaned_data['toc_depth']
-        new_rev.save()
-        new_rev.review_tags.set(*self.cleaned_data['review_tags'])
-        new_rev.localization_tags.set(*self.cleaned_data['localization_tags'])
-        return new_rev
+        Persists the revision and returns it.
+        Takes the view request and document of the revision.
+        Does some specific things when the revision is fully saved.
+        """
+        # have to check for first edit before we save
+        is_first_edit = request.user.wiki_revisions().count() == 0
 
+        # Popping the commit parameter here to make sure there are no surprises
+        kwargs.pop('commit', False)
```

Comment:
This feels wrong to me. Should we be finding the caller that is passing commit=True?  Is there some ```catch TypeError``` at some other level of code?  I'm willing to ignore for another day.